### PR TITLE
feat: Correct requirement categories following top level ISO 25010 standard and change version datatype to custom semantic version datatype

### PR DESCRIPTION
**Pull request by lfvdavid**
Add support for categories that follow top level ISO 25010 standard to requirements and SVCs.

Use Enum classes or similar sollutions.

Change revision/version datatype to a custom semantic version datatype.

### CODE REVIEW

**Comment by lfvdavid** [comment]
Added semantic version datatype to revision/version section of requirements.

Refactored `category` to `categories` and added Enum for categories in the code.

Left to do:

* Update categories in Enum class and json schema, waiting for input on what categories to include.

File: src/reqstool/model_generators/requirements_model_generator.py [hunk 1]
```diff
@@ -289,15 +290,16 @@ def __parse_requirements(self, data):  # NOSONAR
...
+                urn_id = UrnId(urn=urn, id=req["id"])
                 req_data = RequirementData(
-                    id=UrnId(urn=urn, id=req["id"]),
+                    id=urn_id,
                     title=req["title"],
                     significance=SIGNIFANCETYPES(req["significance"]),
                     description=req["description"],
                     rationale=rationale,
-                    category=req["category"],
+                    categories=req["categories"],
```
Comment (lfvjimisola) [inline src/reqstool/model_generators/requirements_model_generator.py:300]: Need to validate input using the relevant enum. You can probably use list comprehension similar to:

`categories = [CATEGORYTYPES[category] for  category in req["categories"]]`

**Review by lfvjimisola — CHANGES_REQUESTED** [review]
* Some feedback given verbally:
** Respect max line length
** Be specific about what in ISO 25010 that we used

File: src/reqstool/model_generators/requirements_model_generator.py (hunk 1 above)
Comment (lfvdavid) [reply src/reqstool/model_generators/requirements_model_generator.py]: Fixed in latest commit

**Review by lfvbarcus — COMMENTED** [review]
I think this looks good, but I'm missing a section in the docs where we link to the ISO standard we are using for the categories so that you can look up when to use which category if it's not clear.

**Comment by lfvdavid** [comment]
> I think this looks good, but I'm missing a section in the docs where we link to the ISO standard we are using for the categories so that you can look up when to use which category if it's not clear.

Added a section under data.adoc about categories and ISO 25010, listing categories and explanations of what each category covers.

**Review by lfvbarcus — APPROVED** [review]

**Review by lfvjimisola — APPROVED** [review]
Nice job!

**Pull request merged**